scene/resources/particles_material.cpp

```diff
@@ -923,6 +1072,34 @@ float ParticlesMaterial::get_param_max(Parameter p_param) const {
 	return params_max[p_param];
 }
 
+void ParticlesMaterial::set_param_direct(ParameterDirect p_param, float p_value) {
```

Comment:
I would prefer these Parameters just have their own setter and getter functions. 

Right now the organization is that values that do not have mins and maxs are set with their own setter/getter functions. e.g. https://github.com/godotengine/godot/blob/ad9a8e70f499267251daaa4cb51f3c2545d4ad4e/scene/resources/particles_material.cpp#L805-L808

If we add a ``set_param_direct()`` users will be confused about which params can go in direct vs min/max. Params should be reserved for things that have a min/max value.

Reply:
I moved `turbulence_noise_strength`, `turbulence_noise_scale` and `turbulence_noise_speed_random` to have their own setter and getter function and like that completely got rid of `set_param_direct()` which could be confusing like you say.